src/IR/IR.ts

```diff
@@ -58,6 +58,7 @@ export interface BaseNode {
   /** type: an uninferrable type, either annotated from the frontend or
    * inserted for language-specific op nodes */
   readonly type?: Type;
+  readonly targetType?: "bigint" | "int" | "string" | "char" | (string & {});
```

Comment:
Nice hack with the `& {}`. I've seen it before, so I remember, but for your sanity might want to write what it does.

Reply:
For languages like C, would it make sense to have separate `"int32"` and `"int64"` target types? Since int64 requires `long` which is one byte longer.

Reply:
> For languages like C, would it make sense to have separate `"int32"` and `"int64"` target types? Since int64 requires `long` which is one byte longer.

Sure, but that can be added when it is first used.